agentarchives/atom/client.py

```diff
@@ -0,0 +1,640 @@
+import json
+import logging
+import os
+import re
+import requests
+try:
+    from urlparse import urlparse, urljoin
+except ImportError:
+    from urllib.parse import urlparse, urljoin
+
+__all__ = ['AtomError', 'ConnectionError', 'AuthenticationError', 'AtomClient']
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AtomError(Exception):
+    pass
+
+
+class ConnectionError(AtomError):
+    pass
+
+
+class AuthenticationError(AtomError):
+    pass
+
+
+class CommunicationError(AtomError):
+    def __init__(self, status_code, response):
+        message = 'AtoM server responded with status code {} (URL: {})'.format(status_code, response.url)
+        self.response = response
+        super(CommunicationError, self).__init__(message)
+
+
+class AtomClient(object):
+    """
+    Client to communicate with a remote AtoM installation using its backend API.
+
+    Note that, while functions follow the same API as the ArchivistsToolkitClient, one major difference is the handling of resource and component IDs.
+    In ArchivistsToolkitClient, resource IDs are longs representing the database row ID.
+    In this client, resource IDs are instead slugs representing the unique part of a record's URL:
+        my-fonds-title
+    This change is due to the fact that slugs are visible by users whereas IDs aren't.
+    """
+
+    RESOURCE = 'resource'
+    RESOURCE_COMPONENT = 'resource_component'
```

Comment:
Does AtoM have a resource / resource component distinction like ArchivesSpace? If not, these should be removed.